### PR TITLE
fix: filtering the data with applying pagination(#806)

### PR DESCRIPTION
**Pull request by js87zz**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description
* The functionality of filtering with paginated data occurs error because of inconsistent `filteredIndex` computed property to observe the filtered data
* related issue(#806)

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: packages/toast-ui.grid/src/store/data.ts [hunk 1]
```diff
@@ -495,12 +495,10 @@ export function create({
...
-        return this.filteredRawData.map(row =>
-          findIndexByRowKey(this, column, id, row.rowKey, false)
-        );
-      }
-      return null;
+      const filteredRawData = this.filteredRawData;
+      return this.filters
+        ? filteredRawData.map(row => findIndexByRowKey(this, column, id, row.rowKey, false))
+        : null;
     },
```
Comment (jung-han) [inline packages/toast-ui.grid/src/store/data.ts]:
```js
get filteredIndex(this: Data) {
  const { filteredRawData, filters } = this;
  return filters
    ? filteredRawData.map(row => findIndexByRowKey(this, column, id, row.rowKey, false))
    : null;
},
```
나

```js
get filteredIndex(this: Data) {
  return this.filters
    ? this.filteredRawData.map(row => findIndexByRowKey(this, column, id, row.rowKey, false))
    : null;
},
```
로 작성되어야 할 것 같습니다!

File: packages/toast-ui.grid/cypress/integration/combinedTest.spec.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
+import { data } from '../../samples/pagination';
```
Comment (jung-han) [inline packages/toast-ui.grid/cypress/integration/combinedTest.spec.ts]: `combinedFeatureTest.spec.ts`는 어떤가요?ㅎㅎ

Comment (js87zz) [reply packages/toast-ui.grid/cypress/integration/combinedTest.spec.ts]: 넵 좋습니다~!ㅋㅋ

File: packages/toast-ui.grid/src/store/data.ts (hunk 1 above)
Comment (js87zz) [reply packages/toast-ui.grid/src/store/data.ts]: 아래처럼 하면 처음에 필터가 적용되지 않은 경우 `filteredRawData`에 대한 observe가 등록되지 않아서 문제가 있더라구요. 위의 예시처럼 변경하겠습니다~!

**Pull request merged**